platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java

```diff
@@ -16,248 +16,40 @@
-        if (applicableAddressBook == null) {
-            // this occurrence was already logged while attempting to determine the applicable address book
-            return false;
-        }
-
-        final NodeId eventCreatorId = event.getHashedData().getCreatorId();
-
-        if (!applicableAddressBook.contains(eventCreatorId)) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(),
-                    "Node {} doesn't exist in applicable address book. Event: {}",
-                    eventCreatorId,
-                    event);
-            return false;
-        }
-
-        final PublicKey publicKey =
-                applicableAddressBook.getAddress(eventCreatorId).getSigPublicKey();
-        if (publicKey == null) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(), "Cannot find publicKey for creator with ID: {}", eventCreatorId);
-            return false;
-        }
-
-        final boolean isSignatureValid = signatureVerifier.verifySignature(
-                event.getHashedData().getHash().getValue(),
-                event.getUnhashedData().getSignature(),
-                publicKey);
-
-        if (!isSignatureValid) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(),
-                    "Event failed signature check. Event: {}, Signature: {}, Hash: {}",
-                    event,
-                    CommonUtils.hex(event.getUnhashedData().getSignature()),
-                    event.getHashedData().getHash());
-        }
-
-        return isSignatureValid;
-    }
+public interface EventSignatureValidator {
 
     /**
      * Validate event signature
      *
      * @param event the event to verify the signature of
      * @return the event if the signature is valid, otherwise null
      */
+    @InputWireLabel("GossipEvent")
     @Nullable
-    public GossipEvent validateSignature(@NonNull final GossipEvent event) {
-        if (nonAncientEventWindow.isAncient(event)) {
-            // ancient events can be safely ignored
-            intakeEventCounter.eventExitedIntakePipeline(event.getSenderId());
-            return null;
-        }
-
-        if (isSignatureValid(event)) {
-            return event;
-        } else {
-            intakeEventCounter.eventExitedIntakePipeline(event.getSenderId());
-            validationFailedAccumulator.update(1);
-
-            return null;
-        }
-    }
+    public GossipEvent validateSignature(@NonNull final GossipEvent event);
 
     /**
      * Set the non-ancient event window that defines the minimum threshold required for an event to be non-ancient
      *
      * @param nonAncientEventWindow the non-ancient event window
      */
-    public void setNonAncientEventWindow(@NonNull final NonAncientEventWindow nonAncientEventWindow) {
-        this.nonAncientEventWindow = Objects.requireNonNull(nonAncientEventWindow);
-    }
+    @InputWireLabel("NonAncientEventWindow")
```

Comment:
This is different than the original. Not necessarily a problem, but want to see the wiring diagram to make sure it still looks ok.

Reply:
wiring diagram added.

Reply:
removing wiring diagram as it creates a merge conflict.